test/end2end_test.go

```diff
@@ -1376,36 +1376,54 @@ func (s) TestDetailedConnectionCloseErrorPropagatesToRpcError(t *testing.T) {
 	close(rpcDoneOnClient)
 }
 
+// TestDetailedGoawayErrorOnGracefulClosePropagatesToRPCError tests the scenario
+// where the configured max connection age expires for a connection. The server
+// is expected to gracefully close the connection in this case by sending a
+// GoAway frame. The test verifies that the error message contained in the
+// GoAway frame is propagated to the RPC error.
 func (s) TestDetailedGoawayErrorOnGracefulClosePropagatesToRPCError(t *testing.T) {
+	// Setup a stub server where the bidi streaming RPC handler blocks on a
+	// channel which is closed when the test is done.
 	rpcDoneOnClient := make(chan struct{})
 	ss := &stubserver.StubServer{
 		FullDuplexCallF: func(stream testpb.TestService_FullDuplexCallServer) error {
 			<-rpcDoneOnClient
 			return status.Error(codes.Internal, "arbitrary status")
 		},
 	}
+	const maxConnectionAge = 100 * time.Millisecond
 	sopts := []grpc.ServerOption{
 		grpc.KeepaliveParams(keepalive.ServerParameters{
-			MaxConnectionAge:      time.Millisecond * 100,
-			MaxConnectionAgeGrace: time.Millisecond,
+			// Use a very short value for max connection age. The server is
+			// expected to send a GoAway once max connection age expires.
+			MaxConnectionAge: maxConnectionAge,
+			// Use a longer value for the grace period. The server hard closes
+			// the connection once the grace period expires, and if this
+			// happens, the client will see an EOF error when attempting to read
+			// from the stream.
```

Comment:
small suggestion to make it even more explicit:

```suggestion
			// from the stream instead of the GOAWAY message expected in this test.
```